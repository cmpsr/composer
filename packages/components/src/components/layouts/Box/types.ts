export { BoxProps } from '@chakra-ui/react';
