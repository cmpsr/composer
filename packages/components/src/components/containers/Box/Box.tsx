import React, { ReactNode } from 'react';
import cn from 'classnames';

export const BOX_DEFAULT_TEST_ID = 'box';

type Props = {
  children?: ReactNode;
  className?: string;
  size?: 'full' | 'screen' | 'content';
  testId?: string;
};

export const Box = ({
  children,
  className,
  size,
  testId = BOX_DEFAULT_TEST_ID,
}: Props) => {
  return (
    <div
      className={cn(
        { 'w-screen h-screen': size === 'screen' },
        { 'w-full h-full': size === 'full' },
        { 'w-full h-full max-w-content mx-auto': size === 'content' },
        className,
      )}
      data-testid={testId}
    >
      {children}
    </div>
  );
};
