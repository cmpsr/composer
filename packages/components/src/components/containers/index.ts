export * from './Box';
export * from './Grid';
export * from './Section';
export * from './SectionChild';
