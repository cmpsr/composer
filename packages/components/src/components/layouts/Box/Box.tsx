import React, { FC } from 'react';
import { Box as ChakraBox } from '@chakra-ui/react';
import { BoxProps } from './types';

export const Box: FC<BoxProps> = (props) => <ChakraBox {...props} />;
