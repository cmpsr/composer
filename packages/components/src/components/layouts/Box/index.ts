export { Box, BoxProps } from '@chakra-ui/react';
