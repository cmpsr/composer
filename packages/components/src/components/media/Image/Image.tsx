import React from 'react';
import { Image as ChakraImage } from '@chakra-ui/react';
import { useResponsiveValue } from '@hooks';
import { ImageProps } from './types';

export const Image = ({ src, ...props }: ImageProps) => {
  const source = useResponsiveValue(src);

  return <ChakraImage src={source} {...props} />;
};
