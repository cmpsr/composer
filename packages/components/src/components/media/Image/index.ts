export { Image, ImageProps } from '@chakra-ui/react';
