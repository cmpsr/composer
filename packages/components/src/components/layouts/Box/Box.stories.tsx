import React from 'react';
import { Meta } from '@storybook/react';
import { Box } from './Box';

export default {
  component: Box,
  title: 'Components/Layouts/Box',
} as Meta;

const Template = (args) => <Box {...args} />;
export const Playground = Template.bind({});
Playground.args = {
  children: "I'm a Box",
  border: '1px solid var(--chakra-colors-primary-default)',
  padding: 'spacer-3',
};
