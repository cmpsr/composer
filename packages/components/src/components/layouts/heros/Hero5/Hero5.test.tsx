import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Hero5 } from './Hero5';
import { BOX_DEFAULT_TEST_ID } from '../../../containers/Box';
import { IMAGE_DEFAULT_TEST_ID } from '../../../primitives/Image';

describe('Hero5', () => {
  const asset = {
    contentType: 'image',
    title: 'Image Title',
    url:
      'https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg',
  };

  it('should match snapshot', () => {
    const hero5 = renderer.create(
      <Hero5
        title="test title"
        style={{}}
        className="foo"
        subtitle="test subtitle"
        desktopAsset={asset}
      />,
    );
    const tree = hero5.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should render children', () => {
    const { getByText } = render(<Hero5 style={{}}>foo</Hero5>);
    const hero5 = getByText('foo');
    expect(hero5).toBeInTheDocument();
  });
  it('Should render with text', () => {
    const { getByText } = render(<Hero5 title="test title" style={{}} />);
    const hero5 = getByText('test title');
    expect(hero5).toBeInTheDocument();
  });
  it('Should render class', () => {
    const { getAllByTestId } = render(
      <Hero5 title="test title" style={{}} className="foo" />,
    );
    const hero5 = getAllByTestId(BOX_DEFAULT_TEST_ID);
    expect(hero5[0].classList).toContain('foo');
  });
  it('Should render desktopAsset', () => {
    const { getByTestId } = render(
      <Hero5
        title="test title"
        style={{}}
        className="foo"
        desktopAsset={asset}
      />,
    );
    const image = getByTestId(IMAGE_DEFAULT_TEST_ID) as HTMLSourceElement;
    expect(image.type).toBe('image/webp');
  });
  it('Should render mobileAsset', () => {
    const { getByTestId } = render(
      <Hero5
        title="test title"
        style={{}}
        className="foo"
        mobileAsset={asset}
      />,
    );
    const image = getByTestId(IMAGE_DEFAULT_TEST_ID) as HTMLSourceElement;
    expect(image.type).toBe('image/webp');
  });
  it('Should render subtitle', () => {
    const { getByText } = render(
      <Hero5
        title="test title"
        subtitle="test subtitle"
        style={{}}
        className="foo"
      />,
    );

    const hero5 = getByText('test subtitle');
    expect(hero5).toBeInTheDocument();
  });
  it('Should render eyebrow', () => {
    const { getByText } = render(
      <Hero5
        title="test title"
        style={{}}
        className="foo"
        eyebrow={'eyebrow test'}
      />,
    );

    const hero5 = getByText('eyebrow test');
    expect(hero5).toBeInTheDocument();
  });
});
