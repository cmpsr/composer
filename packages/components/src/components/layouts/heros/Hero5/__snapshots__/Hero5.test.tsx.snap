// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero5 should match snapshot 1`] = `
<div
  className="w-screen h-screen flex justify-start foo"
  data-testid="box"
>
  <picture
    className="select-none hidden md:block w-1/2 h-full z-0 absolute"
  >
    <source
      className="select-none object-cover w-full h-full"
      data-testid="image"
      srcSet="https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg?fm=webp"
      type="image/webp"
    />
    <img
      alt="Image Title"
      className="select-none object-cover w-full h-full"
      src="https://images.ctfassets.net/o9153kt66j4s/6Ru9unmOhMXrFwKtLV2SRU/7c902fd189741be40f5fef77340cbf1d/placeholder.svg"
    />
  </picture>
  <div
    className="w-full h-full flex justify-center items-center p-8"
    data-testid="box"
  >
    <div
      className="w-full h-full max-w-content mx-auto flex justify-end items-center"
      data-testid="box"
    >
      <div
        className="relative z-10 max-w-hero-copy-small text-right md:text-left"
      >
        <h2
          className="typo-headline-3"
          data-testid="typography"
        >
          test title
        </h2>
        <h6
          className="typo-headline-6 mt-3"
          data-testid="typography"
        >
          test subtitle
        </h6>
        <div
          className="flex justify-end md:justify-start mt-8"
        />
      </div>
    </div>
  </div>
</div>
`;
